src/actions/actions.ts

```diff
@@ -2190,11 +2188,11 @@ class MoveToMatchingBracket extends BaseMovement {
       let matchedPosition: Position | undefined = undefined;
 
       for (const { char, pos } of Position.IterateDocument(position, toFind.nextMatchIsForward)) {
-        if (char === charToMatch) {
+        if (char === charToMatch && charToMatch !== toFind.match) {
           stackHeight++;
         }
 
-        if (char === this.pairings[charToMatch].match) {
+        if (char === toFind.match) {
```

Comment:
This `pairings` bothers me a lot when I implement [MoveToUnclosedRoundBracketBackward](https://github.com/sectioneight/Vim/blob/5a50f2105119bd8cb7ba4a105560a76b2369bb59/src/actions/actions.ts#L2294). And the first hack I did is moving the bracket matching to a separate method called `nextBracket` in https://github.com/vscodevim/Vim/commit/b0a203830c7cda40100c678bd79c2e45c495af94 but now it seems `nextBracket` is no longer a good name as we might match quotes. Something called `nextPairedChar` looks better to me.

Besides, your PR is good enough for this fix but you may want to take a look at [all text objects commands](https://github.com/VSCodeVim/Vim/blob/master/ROADMAP.md#text-objects-only-in-visual-mode-or-after-an-operator), maybe we can make this `abstract class MoveToMatchingBase` more generic?